functions.R

```diff
@@ -881,3 +891,150 @@ plotter <- function(df, variable = NULL, show_labels = TRUE){
   }
 }
 
+# taking factors with one level in osduhs survey data and imputing 'NO' onto it
+relevel_factor_one_lfs <- function(dat_var) {
+  dat_var <- as.character(dat_var)
+  dat_var[is.na(dat_var)] <- 'NO'
+  return(dat_var)
+}
+
+# restructure data types
+restructure_data_types <- function(temp) {
+  for(i in 1:ncol(temp)) {
+    if(grepl('factor', class(temp[, i]))) {
+      temp[, i] <- as.character(temp[, i])
+    } else {
+      temp[, i] <- as.numeric(temp[, i])
+    }
+  }
+  return(temp)
+}
+
+clean_lfs <- function(temp_clean) {
+  
+  # characters back to factors, and everything else numeric
+  temp_clean <- restructure_data_types(temp_clean)
+  
+  # remove extra columns or columns with too many NAs
+  temp_clean$current_student_status_and_type_of_school.1 <- NULL
+  
+  # if the level of a factor is 1, then that factor only has "yes" coded and should replace NA with "NO"
+  temp_clean$job_seeker_checked_wemployers_directly <- relevel_factor_one_lfs(temp_clean$job_seeker_checked_wemployers_directly)
+  temp_clean$job_seeker_checked_wemployment_agency <- relevel_factor_one_lfs(temp_clean$job_seeker_checked_wemployment_agency)
+  temp_clean$jobseeker_contacted_relatives <- relevel_factor_one_lfs(temp_clean$jobseeker_contacted_relatives)
+  temp_clean$jobseeker_looked_at_ads <- relevel_factor_one_lfs(temp_clean$jobseeker_looked_at_ads)
+  temp_clean$jobseeker_placed_or_answered_ads <- relevel_factor_one_lfs(temp_clean$jobseeker_placed_or_answered_ads)
+  temp_clean$jobseeker_other_methods <- relevel_factor_one_lfs(temp_clean$jobseeker_other_methods)
+  
+  # filter to get ontario (should have done this earlier, but missed it)
```

Comment:
What do you mean "should have done this earlier"? If it should be earlier, make it earlier. If it doesn't matter, remove the comment.

Reply:
Sorry, I meant to say I missed that earlier when we went through it the first time, which is irrelevant, so I'll make the comments more clear